### PR TITLE
[DLS] Set data provider features to corresponding connector features

### PR DESCRIPTION
**Pull request by timgrein**
## Closes https://github.com/elastic/enterprise-search-team/issues/4970

This PR sets the data provider features to its corresponding connector features. We need that to adjust the logic inside a data source implementation based on the enabled/disabled features.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/sync_job_runner.py [hunk 1]
```diff
@@ -118,6 +118,12 @@ async def execute(self):
                 await self._sync_done(sync_status=JobStatus.COMPLETED)
                 return
 
+            if hasattr(self.data_provider, "features"):
+                logger.warning(f"{self.data_provider.name} already has 'features' set.")
+            else:
+                logger.debug(f"Setting 'features' for {self.data_provider.name}")
+                self.data_provider.features = self.connector.features
```
Comment (wangch079) [inline connectors/sync_job_runner.py:125]: Should we have a method `set_features` in `BaseDataSource`?

I'm about to introduce a `set_logger` method as well: https://github.com/elastic/connectors-python/pull/806#discussion_r1231472801

Comment (timgrein) [reply connectors/sync_job_runner.py:125]: Makes sense, it's more consistent 👍

**Review by artem-shelkovnikov — APPROVED** [review]
Are you planning to wrap checking features in a function?

Otherwise might make sense to init _features as empty hash/array - depending on the data structure.

**Comment by timgrein** [comment]
> Are you planning to wrap checking features in a function?
> 
> Otherwise might make sense to init _features as empty hash/array - depending on the data structure.

We've an explicit `Features` object, which already provides high-level methods for checking for certain features 👍

**Pull request merged**